force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -110,7 +170,26 @@ public with sharing class FlowControlLoopsCollections {
      * @return The grade for the given score.
      */
     public static String determineGrade(Integer score) {
-        return null; // Replace null with the variable you used to store the result
+        
+        String grade;
+
+        if (score >= 90) {
+            grade = 'A';
+        }
+        else if (score >= 80 && score < 90) {
+            grade = 'B';
+        }
+        else if (score >= 70 && score < 80) {
+            grade = 'C';
+        }
+        else if (score >= 60 && score < 70) {
+            grade = 'D';
+        }
```

Comment:
You don't need to check the upper bounds because your previous checks will handle that. It is not going to get to your first else if it is >90, so there is no reason to check <90 in the conditional.